File: backend/src/main/java/com/pasifcode/syncdoc/domain/entity/Locality.java
package com.pasifcode.syncdoc.domain.entity;

import com.pasifcode.syncdoc.domain.enums.ZoneType;
import jakarta.persistence.*;
import lombok.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Entity
@Table(name = "tb_locality")
public class Locality extends BaseEntity {

    @Column(nullable =false)
    private Long sequenceNumber;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "office_id")
    private Office office;

    @Column(nullable = false, length = 100)
    private String country;

    @Column(length = 100)
    private String region;

    @Column(nullable = false, length = 100)
    private String state;

    @Column(nullable = false, length = 100)
    private String city;

    @Column(length = 100)
    private String district;

    @Column(nullable = false, length = 150)
    private String name;

    @Enumerated(EnumType.STRING)
    private ZoneType zoneType;

    @Column(length = 1000)
    private String observations;

    private Boolean active;

}

File: frontend/src/resources/locality.ts
import axios from "axios";
import { baseUrl } from "utils/requests";

export type ZoneType = "URBAN" | "RURAL";

export type Locality = {
  id?: string;
  sequenceNumber?: number;
  country?: string;
  region?: string;
  state?: string;
  city?: string;
  district?: string;
  name: string;
  zoneType?: ZoneType;
  observations?: string;
  active?: boolean;
};

class LocalityService {
  async create(officeId: string, locality: Locality): Promise<Locality> {
    const response = await axios.post(`${baseUrl}/offices/${officeId}/localities`, locality);
    return response.data;
  }
  async findAllByOffice(officeId: string): Promise<Locality[]> {
    const response = await axios.get(`${baseUrl}/offices/${officeId}/localities`);
    return response.data;
  }
}

export const useLocalityService = () => new LocalityService();